pep-0597.rst

```diff
@@ -33,135 +34,140 @@ Using the default encoding is a common mistake
-other modules using them will emit ``EncodingWarning`` when the
-``encoding`` is omitted.
+When the flag is set, ``io.TextIOWrapper()``, ``open()`` and other
+modules using them will emit ``EncodingWarning`` when the ``encoding``
+argument is omitted.
 
 Since ``EncodingWarning`` is a subclass of ``Warning``, they are
-shown by default, unlike ``DeprecationWarning``.
+shown by default (if the ``encoding_warning`` flag is set), unlike
+``DeprecationWarning``.
 
 
-``encoding="locale"`` option
-----------------------------
+``encoding="locale"``
+---------------------
 
-``io.TextIOWrapper`` accepts ``encoding="locale"`` option. It means
-same to current ``encoding=None``. But ``io.TextIOWrapper`` doesn't
-emit ``EncodingWarning`` when ``encoding="locale"`` is specified.
+``io.TextIOWrapper`` will accept ``"locale"`` as a valid argument to
+``encoding``. It has the same meaning as the current ``encoding=None``,
+except that ``io.TextIOWrapper`` doesn't emit ``EncodingWarning`` when
+``encoding="locale"`` is specified.
 
 
 ``io.text_encoding()``
------------------------
+----------------------
 
-``io.text_encoding()`` is a helper function for functions having
-``encoding=None`` option and passing it to ``io.TextIOWrapper()`` or
+``io.text_encoding()`` is a helper for functions with an
+``encoding=None`` parameter that pass it to ``io.TextIOWrapper()`` or
 ``open()``.
 
-Pure Python implementation will be like this::
+A pure Python implementation will look like this::
 
    def text_encoding(encoding, stacklevel=1):
-       """Helper function to choose the text encoding.
+       """A helper function to choose the text encoding.
 
        When *encoding* is not None, just return it.
-       Otherwise, return the default text encoding (i.e., "locale").
+       Otherwise, return the default text encoding (i.e. "locale").
 
-       This function emits EncodingWarning if *encoding* is None and
-       sys.flags.encoding_warning is true.
+       This function emits an EncodingWarning if *encoding* is None and
+       sys.flags.encoding_warning is True.
 
-       This function can be used in APIs having encoding=None option
-       and pass it to TextIOWrapper or open.
-       But please consider using encoding="utf-8" for new APIs.
+       This function can be used in APIs with an encoding=None parameter
+       that pass it to TextIOWrapper or open.
+       However, please consider using encoding="utf-8" for new APIs.
        """
        if encoding is None:
            if sys.flags.encoding_warning:
                import warnings
-               warnings.warn("'encoding' option is omitted",
-                            EncodingWarning, stacklevel + 2)
+               warnings.warn(
+                   "The 'encoding' argument was omitted; please "
+                   "explicitly specify one (e.g. 'utf-8'), or 'locale' "
+                   "to use the current system's locale encoding",
```

Comment:
```suggestion
                   "The 'encoding' argument was omitted; please "
                   "explicitly specify one (e.g. 'utf-8'), or 'locale' "
                   "to use the current system's locale encoding",
```

If the extra detail here on how to respond to this is not desired/appropriate, everything after the semicolon `;` can be removed

Reply:
Yes. They are removed already in the pull request: https://github.com/python/cpython/pull/19481

Reply:
```suggestion
                   "'encoding' argument not specified.",
```

Thanks, updated to match python/cpython#19481

Reply:
LGTM